xbmc/utils/LangCodeExpander.cpp

```diff
@@ -1790,3 +1870,264 @@ const std::array<ISO3166_1, 245> RegionCodes = {{
     {"zw", "zwe"}
 }};
 // clang-format on
+
+// clang-format off
+// The ISO3166_1 codes above have fewer entries, are in lower-case
+// and lack the name.  Ideally, these should be merged into a
+// single source of truth and the referring functions updated
+// to account for the differences in case.
```

Comment:
This exact thought came to mind looking at this. It looks like we already replicate essentially the same data in multiple tables already. I think a better way to go is to look at what fields are required, and combining into a single table. Just the maintenance of updating 3-4 different lookup tables with the same info seems crazy to me.

Reply:
Whilst I agree with you (I did write the comment after all), I did not feel that I had sufficient knowledge of Kodi to undertake a project of such magnitude.

Reply:
If you've done this much leg work already, you probably have more active knowledge about this specific topic than anyone else right now.

Reply:
@fuzzard - I respectfully disagree.  My 'leg work', as it were, was simply identifying that langinfo.xml was not being interpreted correctly and addressing that.  Language identification/selection is used extensively throughout Kodi.  That is touched on by this PR:

https://github.com/xbmc/xbmc/pull/22716

In it, @fbacher mentioned 36 instances that were found.

Without too much thought, I can think of: the main UI, various closed captions (Live TV, recordings, DVD, other media), EPG, DVD menus, audio track selection, the list goes on.

The focus of my change was very specific to the Python API for addon developers.  The existing functions were not returning the correct information to Python.  I still need to reacquaint myself fully with my code, however, as I recall, the root problem was how the contents of langinfo.xml were being used to generate the language code for Python.

I did my best to balance remediation with backwards compatibility so that existing addons were not greatly impacted but that Python developers could also specifically request some new data if required.

Now that I have (hopefully) addressed the root issue (langinfo.xml interpretation), other developers specialising in other modules requiring language identification/selection can review their code and modify it to take advantage of my update should they choose.

Identifying/updating every single module that touches language identification/selection would be a massive undertaking and a project in its own right.